worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongePlayer.java

```diff
@@ -0,0 +1,227 @@
+import org.spongepowered.api.text.serializer.TextSerializers;
+import org.spongepowered.api.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+import java.util.UUID;
+
+public class SpongePlayer extends AbstractPlayerActor {
+
+    private final Player player;
+
+    protected SpongePlayer(SpongePlatform platform, Player player) {
+        this.player = player;
+        ThreadSafeCache.getInstance().getOnlineIds().add(getUniqueId());
+    }
+
+    @Override
+    public UUID getUniqueId() {
+        return player.getUniqueId();
+    }
+
+    @Override
+    public int getItemInHand() {
+        Optional<ItemStack> is = this.player.getItemInHand();
+        return is.isPresent() ? IDHelper.resolve(is.get().getItem()) : 0;
+    }
+
+    @Override
+    public String getName() {
+        return this.player.getName();
+    }
+
+    @Override
+    public BaseEntity getState() {
+        throw new UnsupportedOperationException("Cannot create a state from this object");
+    }
+
+    @Override
+    public Location getLocation() {
+        org.spongepowered.api.world.Location<World> entityLoc = this.player.getLocation();
+        Vector3d entityRot = this.player.getRotation();
+
+        return SpongeAdapter.adapt(entityLoc, entityRot);
+    }
+
+    @Override
+    public WorldVector getPosition() {
+        Vector3d pos = this.player.getLocation().getPosition();
+        return new WorldVector(LocalWorldAdapter.adapt(SpongeAdapter.adapt(this.player.getWorld())), pos.getX(), pos.getY(), pos.getZ());
+    }
+
+    @Override
+    public com.sk89q.worldedit.world.World getWorld() {
+        return SpongeAdapter.adapt(player.getWorld());
+    }
+
+    @Override
+    public double getPitch() {
+        return getLocation().getPitch();
+    }
+
+    @Override
+    public double getYaw() {
+        return getLocation().getYaw();
+    }
+
+    @Override
+    public void giveItem(int type, int amt) {
+        this.player.getInventory().offer(ItemStack.of(IDHelper.resolveItem(type), amt));
+    }
+
+    @Override
+    public void dispatchCUIEvent(CUIEvent event) {
+        String[] params = event.getParameters();
+        String send = event.getTypeId();
+        if (params.length > 0) {
+            send = send + "|" + StringUtil.joinString(params, "|");
+        }
+    }
+
+    @Override
+    public void printRaw(String msg) {
+        for (String part : msg.split("\n")) {
+            this.player.sendMessage(TextSerializers.LEGACY_FORMATTING_CODE.deserialize(part));
+        }
+    }
+
+    @Override
+    public void printDebug(String msg) {
+        sendColorized(msg, TextColors.GRAY);
+    }
+
+    @Override
+    public void print(String msg) {
+        sendColorized(msg, TextColors.LIGHT_PURPLE);
+    }
+
+    @Override
+    public void printError(String msg) {
+        sendColorized(msg, TextColors.RED);
+    }
+
+    private void sendColorized(String msg, TextColor formatting) {
+        for (String part : msg.split("\n")) {
```

Comment:
This may have implications on formatting though. If you note, it reformats after every new line. There might be some text that relies on that logic, and actually, I believe that even these formatted options may need to use desearalize. I've noticed some oddities.